python/semantic_kernel/functions/kernel_function.py

```diff
@@ -370,27 +428,43 @@ def describe(self) -> KernelFunctionMetadata:
             name=self.name,
             plugin_name=self.plugin_name,
             description=self.description or "",
-            is_semantic=self.is_semantic,
+            is_prompt=self.is_prompt,
             parameters=self.parameters,
         )
 
     async def __call__(
         self,
         kernel: "Kernel",
-        arguments: Optional[KernelArguments] = None,
-        **kwargs: Dict[str, Any],
+        arguments: KernelArguments,
     ) -> "FunctionResult":
-        return await self.invoke(kernel, arguments, **kwargs)
+        """Invoke the function with the given arguments.
+
+        Args:
+            kernel (Kernel): The kernel
+            arguments (KernelArguments): The Kernel arguments
+
+        Returns:
+            FunctionResult: The result of the function
+        """
+        return await self.invoke(kernel, arguments)
 
     async def invoke(
         self,
         kernel: "Kernel",
-        arguments: Optional[KernelArguments] = None,
-        **kwargs: Dict[str, Any],
+        arguments: KernelArguments,
     ) -> "FunctionResult":
-        if not arguments:
-            arguments = KernelArguments(**kwargs)
```

Comment:
this is consistent with dotnet, so let's keep the optional arguments and kwargs and this line, both here and in __call__